src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/CodeResolvingUtils.cs

```diff
@@ -377,5 +377,77 @@ public static bool IsPXSetupBqlCommand(this ITypeSymbol bqlTypeSymbol, PXContext
 				}
 			}
 		}
+
+		/// <summary>
+		/// Returns event handler type for the provided method symbol.
+		/// </summary>
+		/// <param name="symbol">Method symbol for the event handler</param>
+		/// <param name="pxContext">PXContext instance</param>
+		/// <returns>Event Type (e.g. RowSelecting). If method is not an event handler, returns <code>EventType.None</code>.</returns>
+		public static EventType GetEventHandlerType(this IMethodSymbol symbol, PXContext pxContext)
+		{
+			if (symbol.ReturnsVoid && symbol.TypeParameters.IsEmpty && !symbol.Parameters.IsEmpty)
+			{
+				// Loosely check method signature because sometimes business logic from event handler calls is extracted to a separate method
+
+				// Old syntax
+				if (symbol.Parameters[0].Type.OriginalDefinition.InheritsFromOrEquals(pxContext.PXCacheType))
+				{
+					if (symbol.Name == "CacheAttached")
+						return EventType.CacheAttached;
+
+					if (symbol.Parameters.Length >= 2)
+					{
+						switch (symbol.Parameters[1].Type.OriginalDefinition)
```

Comment:
Depth indentation is bigger than 3. It is a violation of our code style)

Reply:
It is exactly 3 levels (3 nested `if`s to avoid two `return EventType.None` statements).

Reply:
3 nested if + switch. Do we consider that switch adds indentation?

Reply:
I think it adds. But I would just split this method in two, each with its own switch, cause it's quite big

Reply:
I think it adds. But I would just split this method in two, each with its own switch, cause it's quite big

Reply:
Actually in this case I'm not sure that it adds 2 levels of indentation. So the code style is not violated, but still two methods each with its own switch could look better